chaco/data_view.py

```diff
@@ -208,10 +208,65 @@ class DataView(OverlayPlotContainer):
     bgcolor = "white"
 
     #: Padding defaults.
-    padding_top = 50
-    padding_bottom = 50
-    padding_left = 50
-    padding_right = 50
+    padding_top = Property(
+        observe='y_axis.[title,orientation], x_axis.[title,orientation]'
+    )
+    padding_bottom = Property(
+        observe='y_axis.[title,orientation], x_axis.[title,orientation]'
+    )
+    padding_left = Property(
+        observe='y_axis.[title,orientation], x_axis.[title,orientation]'
+    )
+    padding_right = Property(
+        observe='y_axis.[title,orientation], x_axis.[title,orientation]'
+    )
+
+    _padding_top = Union(None, Int())
+    _padding_bottom= Union(None, Int())
+    _padding_left = Union(None, Int())
+    _padding_right = Union(None, Int())
```

Comment:
@rahulporuri 
I looked into it and `Int(allow_none=True)` does not work / is not implemented. 

```
>>> from traits.api import HasTraits, Instance, Int
>>> class Thing(HasTraits):
...     a = Int()
...     b = Int(allow_none=True)
...     c = Instance(HasTraits, allow_none=False)
...     d = Instance(HasTraits)  # default value of allow_none for Instance trait type is True
... 
>>> thing = Thing()
>>> thing.d = None
>>> thing.c = None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/aayres/Desktop/traits/traits/base_trait_handler.py", line 75, in error
    object, name, self.full_info(object, name, value), value
traits.trait_errors.TraitError: The 'c' trait of a Thing instance must be a HasTraits, but a value of None <class 'NoneType'> was specified.
>>> thing.b = None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/aayres/Desktop/traits/traits/base_trait_handler.py", line 75, in error
    object, name, self.full_info(object, name, value), value
traits.trait_errors.TraitError: The 'b' trait of a Thing instance must be an integer, but a value of None <class 'NoneType'> was specified.
```

Reply:
I think this deserves an issue in traits - traits should explicitly complain during class definition instead of at runtime.

Reply:
That makes sense, for my own understanding we want to catch a user setting `allow_none` in a trait definition correct?  I suspect, since `allow_none` is not a supported arg/kwarg for the `Int` trait type, it is now being used as metadata? Ie one could do something like `@observe('+allow_none')`?
We should probably special handle this case